### PR TITLE
Add RSpec model/request tests to rails preset

### PR DESCRIPTION
**Pull request by jonathan-raines**
Added Rspec model and request tests to the Rails present for models and controllers

### CODE REVIEW

**Comment by otavioschwanck** [comment]
> Added Rspec model and request tests to the Rails present for models and controllers

maybe you could add this globally:
```lua
{ "app/(.*).rb", { { "spec/[1]_spec.rb", "Test" } },
{ "spec/(.*).rb", { { "app/[1]_spec.rb", "Original" } }
```

and for the tests with different relative spec file, add manually (like requests)

File: lua/telescope-alternate/presets.lua [hunk 1]
```diff
@@ -1,34 +1,51 @@
...
-    { "app/models/**/*[2:singularize].rb", "Model" },
-    { "app/views/[1][2]/*.html.erb", "View" },
-    { "app/helpers/**/*[2]_helper.rb", "Helper" },
-    { "app/serializers/**/*[2]_serializer.rb", "Serializer" }
+  { 'app/controllers(.*)/(.*)_controller.rb', {
+    { 'app/models/**/*[2:singularize].rb', 'Model' },
+    { 'app/views/[1][2]/*.html.erb', 'View' },
+    { 'app/helpers/**/*[2]_helper.rb', 'Helper' },
+    { 'app/serializers/**/*[2]_serializer.rb', 'Serializer' },
+    { 'spec/requests[1]/[2]*_spec.rb', 'Request Test' },
```
Comment (otavioschwanck) [inline lua/telescope-alternate/presets.lua]: Leave all spec relates to the rspec preset.  So who uses minitest can do differently

File: lua/telescope-alternate/presets.lua [hunk 2]
```diff
@@ -1,34 +1,51 @@
...
-    { "app/helpers/**/*[1]_helper.rb", "Helper" },
+  { 'app/views/(.*)/(.*).html.(.*)', {
+    { 'app/controllers/**/*[1]_controller.rb', 'Controller' },
+    { 'app/models/[1:singularize].rb', 'Model' },
+    { 'app/helpers/**/*[1]_helper.rb', 'Helper' },
+  } },
+}
+
+M.rspec = {
+  { 'app/models/(.*).rb', {
```
Comment (otavioschwanck) [inline lua/telescope-alternate/presets.lua]: Maybe something like that?

{ "app/(.*).rb", { { "spec/[1]_spec.rb", "Test" } },
{ "spec/(.*).rb", { { "app/[1]_spec.rb", "Original" } }

And about the request tests, maybe is more relate to the controller only.

**Review by otavioschwanck — CHANGES_REQUESTED** [review]

**Comment by jonathan-raines** [comment]
Yeah I was starting to think an rspec preset would be better. I will get to work on it

**Comment by jonathan-raines** [comment]
This more verbose than I'd like. Rspec considers `spec/controllers` to be obsolete to `spec/requests`, which doesn't play nice with the wildcard regex. I've been trying to figure out to have one preset that excludes controllers, which would generate most of the needed alternate files, then another preset specifically to handle the `controllers -> requests` & `requests -> controllers` case.

If you have any ideas, please let me know!

File: lua/telescope-alternate/presets.lua [hunk 3]
```diff
@@ -20,8 +20,41 @@ M.rails = {
...
+  { "spec/services/(.*)_spec.rb", {
+    { "app/services/[1].rb", "Original" },
+  } },
+  { "app/mailers/(.*).rb", {
+    { "spec/mailers/[1]_spec.rb", "Test" }
+  } },
+  { "spec/mailers/(.*)_spec.rb", {
+    { "app/mailers/[1].rb", "Original" },
+  } },
+}
```
Comment (otavioschwanck) [inline lua/telescope-alternate/presets.lua]:
```suggestion
M.rspec = {
  { "app/(.*).rb", { { "spec/[1]_spec.rb", "Test" } } },
  { "app/controllers/(.*)_controller.rb", {
    { "spec/requests/[1]_spec.rb", "Request Test" }
  } },
  { "spec/requests/(.*)_spec.rb", {
    { "app/controllers/[1]_controller.rb", "Controller Test" },
  } },
}
```

**Review by otavioschwanck — CHANGES_REQUESTED** [review]

**Comment by jonathan-raines** [comment]
The only issue I see with this implementation is going from a controllers test back to the original. `{ 'spec/(.*)_spec.rb', { { 'app/[1].rb', 'Original' } } }` catches and provides option for a new Original file for `app/requests/...`.

image

The verbose implementation doesn't encounter that, but it is more verbose lol

**Comment by otavioschwanck** [comment]
> The only issue I see with this implementation is going from a controllers test back to the original. `{ 'spec/(.*)_spec.rb', { { 'app/[1].rb', 'Original' } } }` catches and provides option for a new Original file for `app/requests/...`.
> 
> image
> 
> The verbose implementation doesn't encounter that, but it is more verbose lol

i understand your point, but with the more generic approach, we can use other structures, like on my company, we have cells/ and business/.

So i think is ok to have the new there, the NEW is always above the matches

**Comment by otavioschwanck** [comment]
@jonathan-raines thanks for your contribution!

**Pull request merged**